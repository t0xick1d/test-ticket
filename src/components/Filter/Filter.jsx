import React from 'react';
import ToggleButton from '@mui/material/ToggleButton';
import ToggleButtonGroup from '@mui/material/ToggleButtonGroup';

const Filter = ({ alignment, handleChange }) => {
   return (
      <>
         <ToggleButtonGroup
            color="primary"
            value={alignment}
            exclusive
            onChange={handleChange}
            aria-label="Platform"
         >
            <ToggleButton value="cheap">Самый дешевый</ToggleButton>
            <ToggleButton value="speed">Самый Быстрый</ToggleButton>
         </ToggleButtonGroup>
      </>
   );
};

export default Filter;
