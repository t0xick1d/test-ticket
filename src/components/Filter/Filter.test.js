import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Filter from './Filter';

describe('Filter snapshot', () => {
   const mockHandleChange = jest.fn();
   beforeEach(() => {
      mockHandleChange.mockClear();
   });

   it('renders both toggle buttons with correct text', () => {
      render(<Filter alignment="cheap" handleChange={mockHandleChange} />);
      expect(screen.getByText('Самый дешевый')).toBeInTheDocument();
      expect(screen.getByText('Самый Быстрый')).toBeInTheDocument();
   });

   it('calls handleChange when a button is clicked', () => {
      render(<Filter alignment="cheap" handleChange={mockHandleChange} />);
      fireEvent.click(screen.getByText('Самый Быстрый'));
      expect(mockHandleChange).toHaveBeenCalledTimes(1);
   });

   it('activates the correct button based on alignment prop', () => {
      const { rerender } = render(<Filter alignment="cheap" handleChange={mockHandleChange} />);
      const cheapButton = screen.getByText('Самый дешевый');
      const speedButton = screen.getByText('Самый Быстрый');

      expect(cheapButton.getAttribute('aria-pressed')).toBe('true');
      expect(speedButton.getAttribute('aria-pressed')).toBe('false');

      rerender(<Filter alignment="speed" handleChange={mockHandleChange} />);

      expect(cheapButton.getAttribute('aria-pressed')).toBe('false');
      expect(speedButton.getAttribute('aria-pressed')).toBe('true');
   });

   it('matches snapshot', () => {
      const { asFragment } = render(<Filter alignment="cheap" handleChange={mockHandleChange} />);
      expect(asFragment()).toMatchSnapshot();
   });
});
