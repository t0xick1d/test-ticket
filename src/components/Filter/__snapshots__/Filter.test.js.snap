// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter snapshot matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="Platform"
    class="MuiToggleButtonGroup-root MuiToggleButtonGroup-horizontal css-1qb8o-MuiToggleButtonGroup-root"
    role="group"
  >
    <button
      aria-pressed="true"
      class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-firstButton css-w4gp61-MuiButtonBase-root-MuiToggleButton-root"
      tabindex="0"
      type="button"
      value="cheap"
    >
      Самый дешевый
    </button>
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-lastButton css-w4gp61-MuiButtonBase-root-MuiToggleButton-root"
      tabindex="0"
      type="button"
      value="speed"
    >
      Самый Быстрый
    </button>
  </div>
</DocumentFragment>
`;

exports[`Filter snapshot matches snapshot 2`] = `
<DocumentFragment>
  <div
    aria-label="Platform"
    class="MuiToggleButtonGroup-root MuiToggleButtonGroup-horizontal css-1qb8o-MuiToggleButtonGroup-root"
    role="group"
  >
    <button
      aria-pressed="true"
      class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-firstButton css-w4gp61-MuiButtonBase-root-MuiToggleButton-root"
      tabindex="0"
      type="button"
      value="cheap"
    >
      Самый дешевый
    </button>
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-lastButton css-w4gp61-MuiButtonBase-root-MuiToggleButton-root"
      tabindex="0"
      type="button"
      value="speed"
    >
      Самый Быстрый
    </button>
  </div>
</DocumentFragment>
`;

exports[`SPINNER TEST 1`] = `
<button
  aria-pressed="false"
  class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-firstButton css-w4gp61-MuiButtonBase-root-MuiToggleButton-root"
  tabindex="0"
  type="button"
  value="cheap"
>
  Самый дешевый
</button>
`;

exports[`SPINNER TEST 2`] = `
<button
  aria-pressed="false"
  class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-lastButton css-w4gp61-MuiButtonBase-root-MuiToggleButton-root"
  tabindex="0"
  type="button"
  value="speed"
>
  Самый Быстрый
</button>
`;
